### PR TITLE
Fix environment files defect in daemon mode

### PR DESCRIPTION
**Pull request by awharn**
Fixes behavior where, in daemon mode, the daemon client's provided environment variables override and clobber the environment variables set from the environment file.

### CODE REVIEW

**Review by ATorrise — APPROVED** [review]

**Review by KevinLoesch1 — APPROVED** [review]

File: packages/cmd/__tests__/CommandProcessor.test.ts [hunk 1]
```diff
@@ -1744,6 +1746,75 @@ describe("Command Processor", () => {
...
+            },
+            silent: true
+        };
+        EnvFileUtils.setEnvironmentForApp("zowe", false);
+        process.env.UNIT_TEST_ENV = "old";
+        try {
+            const envVarCount = Object.keys(process.env).length;
+            const commandResponse: ICommandResponse = await processor.invoke(parms);
+            expect(commandResponse).toBeDefined();
+            expect(commandResponse).toMatchSnapshot();
```
Comment (zFernand0) [inline packages/cmd/__tests__/CommandProcessor.test.ts:1803]: I'm curious what the purpose of using snapshots here is. It looks like the command response is just an empty object?

Not requesting changes as there are other tests in this file that take snapshots for the same exact empty command response object

**Review by zFernand0 — APPROVED** [review]
Changes LGTM! :yum:

File: packages/cmd/__tests__/CommandProcessor.test.ts (hunk 1 above)
Comment (awharn) [reply packages/cmd/__tests__/CommandProcessor.test.ts:1803]: I just copied it from the above test, I am not sure either. I can remove it if you prefer.

Comment (zFernand0) [reply packages/cmd/__tests__/CommandProcessor.test.ts:1803]: no need 😋

**Pull request merged**